sklearn/manifold/t_sne.py

```diff
@@ -423,18 +423,30 @@ def trustworthiness(X, X_embedded, n_neighbors=5, precomputed=False):
         dist_X = X
     else:
         dist_X = pairwise_distances(X, squared=True)
-    ind_X = np.argsort(dist_X, axis=1)
+    np.fill_diagonal(dist_X, np.inf)  # excludes dist(x_i, x_i)
```

Comment:
We should not modify `X` inplace in the precomputed case.
Adding a copy would solve this problem.

Reply:
Done